docs/make_work.jl

```diff
@@ -34,9 +34,14 @@ Base.walkdir(str::String) = Base.walkdir(str; follow_symlinks=true)
 
 # When we read a `doc.main` from an experimental package, we need to equip all
 # its entries with a prefix to fit with our docs.
-setup_experimental_prefix(docs::String, prefix::String) = prefix * docs
-setup_experimental_prefix(docs::Pair{String, Vector{T}}, prefix::String) where T = Pair{String, Vector{T}}(docs[1], setup_experimental_prefix(docs[2], prefix))
-setup_experimental_prefix(docs::Vector{T}, prefix::String) where T = T[setup_experimental_prefix(entry, prefix) for entry in docs]
+function setup_experimental_prefix(Oscar::Module, docs::String, prefix::String)
```

Comment:
Meta comment: I find `setup_experimental_prefix` rather opaque. I have no idea what typical inputs might be and what it does, and the comment before is not really helping me either.

Perhaps we could have some example calls / outputs in a comment before it. Perhaps it would also help if this code was a single plain function instead of being "cleverly" spread over three methods.

Reply:
I added some examples what these methods accomplish together. I could also turn it into one function, but I would make it recursive (which it also is now) and it would use a bunch of `typeof` calls. Once you look at the examples, let me know what you think. Maybe these are already fine as an explanation.